rosidl_runtime_py/set_message.py

```diff
@@ -33,9 +33,15 @@ def set_message_fields(msg: Any, values: Dict[str, str]) -> None:
     :param values: The values to set in the ROS message. The keys of the dictionary represent
         fields of the message.
     :raises AttributeError: If the message does not have a field provided in the input dictionary.
-    :raises ValueError: If a message value does not match its field type.
+    :raises TypeError: If a message value does not match its field type.
```

Comment:
`ValueError` is incorrect here. If the value has an undesired type the correct exception type is `TypeError`. `ValueError` should only be used if the value type is correct but the value is e.g. outside a specific range.